### DISCUSSION TITLE
Quarto from Doxygen XML

### DISCUSSION THREAD

**Discussion by jdc-pub** [quarto-dev/quarto-cli#10293] [Q&A] [answered]
I haven't used it myself, but I understand Doxygen's XML output is a useful intermediate representation for newer documentation website templates. Has anyone tried integrating Doxygen with Quarto through an extension? I've seen a lot of progress in documenting R, Python, and Julia packages with Quarto, and I think Doxygen's XML might be useful when trying to document C++ projects with Quarto.

**Comment by mcanouil** [comment]
It does not seem to be the same thing.
Julia/R/Python are only used to compute tables, figures, etc. (as Markdown or raw HTML/LaTeX).
The actual input in Quarto is Markdown.

In case, you were talking about using Quarto to generate the documentation for those languages uses Quarto (not the other way around):
- some work on `pkgdown` for R
- quartodoc for Python
- not aware of any initiative for Julia

**Reply by jdc-pub** [reply]
I'm more thinking about an extension that takes in a Doxygen XML file as an input, and outputs appropriately cross-referenced callout blocks.

**Reply by mcanouil** [reply, marked as answer]
I don't think a Quarto extension is the right way or best way, because what you need to do is a big processing of the Doxygen XML file which has nothing to do with Quarto. Once that processing is done and you have markdown files structured possibly as a Quarto website project, use Quarto.

You could do that in a pre-render script eventually.
